app/services/legacy/dataset_import_service.rb

```diff
@@ -0,0 +1,307 @@
+  #     datafile.save!(validate: false)
+  #   end
+  #
+  #   datafile.uuid = resource["id"]
+  #   datafile.format = resource["format"]
+  #   datafile.name = resource["description"]
+  #   datafile.name = "No name specified" if datafile.name.strip() == ""
+  #   datafile.created_at = dataset.created_at
+  #   datafile.updated_at = dataset.last_updated_at
+  #
+  #   if !resource["date"].blank? && !documentation?(resource['format'])
+  #     dates = get_start_end_date(resource["date"])
+  #     if dataset.frequency != 'never'
+  #       begin
+  #         datafile.start_date = Date.parse(dates[0])
+  #       rescue ArgumentError
+  #         datafile.start_date = Date.new(1,1,1)
+  #       end
+  #       begin
+  #         datafile.end_date = Date.parse(dates[1])
+  #       rescue ArgumentError
+  #         datafile.end_date = Date.new(1,1,1)
+  #       end
+  #     end
+  #   end
+  #
+  #   datafile.save!(validate: false)
+  # end
+
+  def add_inspire_metadata(dataset_id, dataset)
+    extras = dataset["extras"]
+
+    inspire = InspireDataset.find_or_create_by(dataset_id: dataset_id)
+    inspire.bbox_east_long = get_extra(extras, 'bbox-east-long')
+    inspire.bbox_north_lat = get_extra(extras, 'bbox-north-lat')
+    inspire.bbox_south_lat = get_extra(extras, 'bbox-south-lat')
+    inspire.bbox_west_long = get_extra(extras, 'bbox-west-long')
+    inspire.coupled_resource = get_extra(extras, 'coupled-resource')
+    inspire.dataset_reference_date = get_extra(extras, 'dataset-reference-date')
+    inspire.frequency_of_update = get_extra(extras, 'frequency-of-update')
+    inspire.harvest_object_id = get_extra(extras, 'harvest_object_id')
+    inspire.harvest_source_reference = get_extra(extras, 'harvest_source_reference')
+    inspire.import_source = get_extra(extras, 'import_source')
+    inspire.metadata_date = get_extra(extras, 'metadata-date')
+    inspire.metadata_language = get_extra(extras, 'metadata-language')
+    inspire.provider = get_extra(extras, 'provider')
+    inspire.resource_type = get_extra(extras, 'resource-type')
+    inspire.responsible_party = get_extra(extras, 'responsible-party')
+    inspire.spatial = get_extra(extras, 'spatial')
+    inspire.spatial_data_service_type = get_extra(extras, 'spatial-data-service-type')
+    inspire.spatial_reference_system = get_extra(extras, 'spatial-reference-system')
+    inspire.guid = get_extra(extras, 'guid')
+    inspire
+  end
+
+  # Generates a summary from the provided notes as in
+  # the legacy metadata we don't have a summary field.
+  def generate_summary(notes)
+    return notes if notes && notes != ""
+    "No description provided"
+  end
+
+  def build_licence
+    return 'no-licence' if licence.blank?
+    return 'other' if licence != "uk-ogl"
+    licence
+  end
+
+  def build_licence_other
+    return licence if licence != "uk-ogl"
+  end
+
+  # Converts a legacy frequency into a new-style frequency
+  def build_frequency
```

Comment:
You've changed the behaviour here: the old code makes sure if any resource has no date, then the dataset frequency is set to `never`. Your method now omits this.

Reply:
Hi @laurentqro - this is why some datasets were not displaying correctly - specifically datasets that had time series data and additional data. Frequencies were being mutated to 'never' even if the dataset was intentionally a time series dataset. This code is not accounting for non time series datasets though - going to investigate next